src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -1878,11 +1878,14 @@ public async Task PostAsync_CallMethod_EmptyContent(Uri remoteServer)
         [InlineData(false, "1.1")]
         [InlineData(true, "1.1")]
         [InlineData(null, "1.1")]
+        [InlineData(false, "2.0")]
+        [InlineData(true, "2.0")]
+        [InlineData(null, "2.0")]
         public async Task PostAsync_ExpectContinue_Success(bool? expectContinue, string version)
         {
             using (HttpClient client = CreateHttpClient())
             {
-                var req = new HttpRequestMessage(HttpMethod.Post, Configuration.Http.RemoteEchoServer)
+                var req = new HttpRequestMessage(HttpMethod.Post, version == "2.0" ? Configuration.Http.Http2RemoteEchoServer : Configuration.Http.RemoteEchoServer)
```

Comment:
The remote Azure echo servers don't really do anything with 'Expect: 100-continue'.  They never will send back a '100 Continue' HTTP response status code.

Normally, the whole point of even using 'Expect: 100-continue' semantics is to help avoid POST'ing a large request body to the server in case the server demand authentication.  So, using this feature allows the client to wait before sending a large request body before getting denied with a '401 Unauthorized' for example.

So, I'm not clear about the purpose of these tests here using the remote Azure echo servers.  At most, it is testing for nothing bad happening when sending 'Expect: 100-continue' and not getting any 100 Continue response. 

It might be useful to add comments to this test to explain that. Otherwise, people reading this code might assume that it is testing something that it really isn't.

Reply:
I'm not 100% sure either but #34460 asked for updating all existing tests to external servers to also use HTTP/2. As minimum we will get validation that we can interoperate with real server and not just loopback.

Reply:
ok

Reply:
100 Continue responses are automatic as soon as you read the request body. I assume the echo server does that?

Reply:
That's a great question.  I'm not sure if the echo server really support 100 Continue at all.

The "echo" server is running as an ASP.NET Core 2.2 server running in Azure App Service:
https://github.com/davidsh/corefx-net-endpoint/tree/master/NetCoreServer.  

I think the default of ASP.NET is IIS?  Does that support 100 Continue by default?

Reply:
Yes, I think all MS servers support 100 Continue. It's implemented in Http.Sys so HttpListener, IIS, and HttpSys (WebListener) all get it for free. Kestrel also supports it. There is no application code required to send 100 Continue, only reading from the request body.